packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -29,7 +29,11 @@ export function importCsjOrEsModule(filePath: string): any {
     const imported = require(filePath);
     return imported.default !== undefined ? imported.default : imported;
   } catch (e: any) {
-    if (e.code === "ERR_REQUIRE_ESM") {
+    if (
+      e.code === "ERR_REQUIRE_ESM" ||
+      e.message === "module is not defined" ||
```

Comment:
Do you mind adding a comment about when these cases happen? @fvictorio

Reply:
Other than that, LGTM

Reply:
I realized that this fix, while useful, has nothing to do with the latest dependencies CI job. The problem is that I cloned a fresh Hardhat repo, deleted the lockfile, and ran the tests, and got a failure for the test that checks this. But this failed because I was using node v22; the error in the CI was different :man_facepalming:

I still think having this is useful, and double-checked manually that it's necessary, although I don't understand why this doesn't fail when the CI is run with v22.

Anyway, the latest dependencies job was failing because solhint published a new version that makes our "no-errors" fixture project have an error. I fixed that in this PR too.

Reply:
ouch